packages/proxyIdentity/src/ProxyManager.ts

```diff
@@ -2,25 +2,34 @@
 /* eslint-disable @typescript-eslint/explicit-function-return-type */
 /* eslint-disable no-useless-constructor */
 import { Signer, Contract, ContractFactory } from 'ethers';
+import { Provider } from 'ethers/providers';
 import { abi as proxyAbi, bytecode as proxyBytecode } from '../build/contracts/ProxyIdentity.json';
 import { abi as erc1155Abi } from '../build/contracts/ERC1155Multiproxy.json';
 
 export class ProxyManager {
   private proxyFactory: ContractFactory;
 
+  private provider: Provider;
+
   constructor(private erc1056: string, private erc1155: string, private owner: Signer) {
     this.proxyFactory = new ContractFactory(proxyAbi, proxyBytecode, this.owner);
+    this.provider = owner.provider;
   }
 
   async createProxy(serial: string): Promise<Contract> {
+    const address = await this.proxyFactory.signer.getAddress();
     return this.proxyFactory.deploy(
       this.erc1056, this.erc1155, serial, await this.owner.getAddress(),
+      { nonce: await this.provider.getTransactionCount(address) },
```

Comment:
I hope this is solving this problem. Otherwise blockHash of the recent block could be an option.

Reply:
yes. this seems to be working. I consoled tx nonces and they are indeed different